src/main/java/org/damap/base/rest/storage/InternalStorageTranslationValidator.java

```diff
@@ -0,0 +1,79 @@
+package org.damap.base.rest.storage;
+
+import jakarta.ws.rs.ClientErrorException;
+import jakarta.ws.rs.NotFoundException;
+import jakarta.ws.rs.core.Response;
+import lombok.experimental.UtilityClass;
+import lombok.extern.jbosslog.JBossLog;
+import org.damap.base.domain.InternalStorage;
+import org.damap.base.domain.InternalStorageTranslation;
+import org.damap.base.repo.InternalStorageRepo;
+import org.damap.base.repo.InternalStorageTranslationRepo;
+
+@UtilityClass
+@JBossLog
+public class InternalStorageTranslationValidator {
+
+  public void validateForCreation(
+      InternalStorageTranslationDO internalStorageTranslationDO,
+      InternalStorageRepo internalStorageRepo,
+      InternalStorageTranslationRepo internalStorageTranslationRepo)
+      throws ClientErrorException {
+    log.info("Validating internal storage translation for creation");
+
+    // Check if no translation for the same language exists
+    if (internalStorageTranslationRepo.existsTranslationForStorageIdAndLanguageCode(
+        internalStorageTranslationDO.getStorageId(),
+        internalStorageTranslationDO.getLanguageCode())) {
+      throw new ClientErrorException(
+          "Translation for language code "
+              + internalStorageTranslationDO.getLanguageCode()
+              + " already exists",
+          Response.Status.BAD_REQUEST);
+    }
+
+    InternalStorageTranslationValidator.validationCommon(
+        internalStorageTranslationDO, internalStorageRepo);
+  }
+
+  public void validateForUpdate(
+      String id,
+      InternalStorageTranslationDO internalStorageTranslationDO,
+      InternalStorageTranslationRepo internalStorageTranslationRepo,
+      InternalStorageRepo internalStorageRepo)
+      throws ClientErrorException {
+    log.info("Validating internal storage translation for update");
+
+    InternalStorageTranslation internalStorageTranslation =
+        internalStorageTranslationRepo.findById(Long.parseLong(id));
+
+    if (internalStorageTranslation == null) {
+      throw new NotFoundException("No internal storage translation with ID " + id + " found");
+    }
+
+    InternalStorageTranslationValidator.validationCommon(
+        internalStorageTranslationDO, internalStorageRepo);
+  }
```

Comment:
major: missing validation to check if another translation with the same language key already exists.